qiskit/utils/classtools.py

```diff
@@ -28,7 +28,7 @@
 # `type` itself has several methods (mostly dunders).  When we are wrapping those names, we need to
 # make sure that we don't interfere with `type.__getattribute__`'s handling that circumvents the
 # normal inheritance rules when appropriate.
-_TYPE_METHODS = set(dir(type))
+_TYPE_METHODS = set(dir(type)) - {"__init_subclass__"}
```

Comment:
I have been thinking about if there is a comment that should go with the change here, but I am not sure if I know how `__init_subclass__` is handled well enough to write that comment. I was also trying to decide if `__init_subclass__` was the only thing that should go here. It's at least the only thing that matters to the tests.